src/_pytest/outcomes.py

```diff
@@ -77,132 +73,118 @@ def __init__(
-        to skip a doctest statically.
-    """
-    __tracebackhide__ = True
-    raise Skipped(msg=reason, allow_module_level=allow_module_level)
+    def __call__(self, reason: str = "", pytrace: bool = True) -> NoReturn:
+        """Explicitly fail an executing test with the given message.
 
+        :param reason:
+            The message to show the user as reason for the failure.
 
-@_with_exception(Failed)
-def fail(reason: str = "", pytrace: bool = True) -> NoReturn:
-    """Explicitly fail an executing test with the given message.
+        :param pytrace:
+            If False, msg represents the full failure information and no
+            python traceback will be reported.
 
-    :param reason:
-        The message to show the user as reason for the failure.
+        :raises pytest.fail.Exception:
+            The exception that is raised.
+        """
+        __tracebackhide__ = True
+        raise Failed(msg=reason, pytrace=pytrace)
 
-    :param pytrace:
-        If False, msg represents the full failure information and no
-        python traceback will be reported.
 
-    :raises pytest.fail.Exception:
-        The exception that is raised.
-    """
-    __tracebackhide__ = True
-    raise Failed(msg=reason, pytrace=pytrace)
+class _XFail:
+    Exception: type[XFailed] = XFailed
 
+    def __call__(self, reason: str = "") -> NoReturn:
+        """Imperatively xfail an executing test or setup function with the given reason.
 
-class XFailed(Failed):
-    """Raised from an explicit call to pytest.xfail()."""
+        This function should be called only during testing (setup, call or teardown).
 
+        No other code is executed after using ``xfail()`` (it is implemented
+        internally by raising an exception).
 
-@_with_exception(XFailed)
-def xfail(reason: str = "") -> NoReturn:
-    """Imperatively xfail an executing test or setup function with the given reason.
+        :param reason:
+            The message to show the user as reason for the xfail.
 
-    This function should be called only during testing (setup, call or teardown).
+        .. note::
+            It is better to use the :ref:`pytest.mark.xfail ref` marker when
+            possible to declare a test to be xfailed under certain conditions
+            like known bugs or missing features.
 
-    No other code is executed after using ``xfail()`` (it is implemented
-    internally by raising an exception).
+        :raises pytest.xfail.Exception:
+            The exception that is raised.
+        """
+        __tracebackhide__ = True
+        raise XFailed(msg=reason)
 
-    :param reason:
-        The message to show the user as reason for the xfail.
 
-    .. note::
-        It is better to use the :ref:`pytest.mark.xfail ref` marker when
-        possible to declare a test to be xfailed under certain conditions
-        like known bugs or missing features.
+# Exposed helper methods.
```

Comment:
My slight preference it to have the `exit = _Exit` etc. below the `class _Exit` instead of group at the bottom, it makes it easier to understand what it's about IMO.